lib/ClientImpl.cc

```diff
@@ -222,12 +222,12 @@ void ClientImpl::handleCreateProducer(const Result result, const LookupDataResul
 void ClientImpl::handleProducerCreated(Result result, ProducerImplBaseWeakPtr producerBaseWeakPtr,
                                        CreateProducerCallback callback, ProducerImplBasePtr producer) {
     if (result == ResultOk) {
-        auto pair = producers_.emplace(producer.get(), producer);
-        if (!pair.second) {
-            auto existingProducer = pair.first->second.lock();
+        auto address = producer.get();
+        auto existingProducer = producers_.create(address, producer);
```

Comment:
`create()` is a bit confusing here as a name. I was wondering why do we need to "create a new consumer here"... 

Could we rename the method as `putIfAbsent()`?

Reply:
Yes, it makes sense.

Reply:
I renamed these methods as `put` and `putIfAbsent` now.